ls1intum/Artemis
# `General`: Apply filters to next relevant exercise

### Checklist
#### General


#### Client

### Motivation and Context



The filters of the course exercise view are currently not applied to the next relevant exercise shown at the top.
<img width="1176" alt="image (1)" src="https://user-images.githubusercontent.com/23171488/161755369-a84a008e-07df-4202-8778-91f16b482e0a.png">

### Description


This PR applies the filter to the selection algorithm of the "next relevant exercise" as well by only selecting from those exercises that fulfill the selected filter.

### Steps for Testing

Prerequisites:
- 1 Student
- A few exercises in different, filterable stages: Optional, with full points, ... that are candidates to appear as next relevant exercise

--> Select and deselect filters from the filter dropdown and make sure that only exercises that fulfill the currently selected filters appear as "next relevant exercise".

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-exercises.component.ts | 77.98% | 98.81% |

### Screenshots



https://user-images.githubusercontent.com/23171488/161756203-2f6a2b82-d99d-4aab-bbd9-577068ea001c.mov
